File: library-core/src/main/java/com/mikepenz/fastadapter/IItemList.java
package com.mikepenz.fastadapter;

import java.util.List;

import javax.annotation.Nullable;

/**
 * The Item list interface
 */

public interface IItemList<Item> {

    int getAdapterPosition(long identifier);

    void remove(int position, int preItemCount);

    void removeRange(int position, int itemCount, int preItemCount);

    void move(int fromPosition, int toPosition, int preItemCount);

    int size();

    void clear(int preItemCount);

    boolean isEmpty();

    void addAll(List<Item> items, int preItemCount);

    void set(int position, Item item, int preItemCount);

    void set(List<Item> items, int preItemCount, @Nullable IAdapterNotifier adapterNotifier);

    void setNewList(List<Item> items, boolean notify);

    void addAll(int position, List<Item> items, int preItemCount);

    List<Item> getItems();

    Item get(int position);
}

File: library-core/src/main/java/com/mikepenz/fastadapter/IItemList.kt
package com.mikepenz.fastadapter

/**
 * The Item list interface
 */

interface IItemList<Item : GenericItem> {

    var active: Boolean

    val isEmpty: Boolean

    val items: MutableList<Item>

    fun getAdapterPosition(identifier: Long): Int

    fun remove(position: Int, preItemCount: Int)

    fun removeRange(position: Int, itemCount: Int, preItemCount: Int)

    fun move(fromPosition: Int, toPosition: Int, preItemCount: Int)

    fun size(): Int

    fun clear(preItemCount: Int)

    fun addAll(items: List<Item>, preItemCount: Int)

    operator fun set(position: Int, item: Item, preItemCount: Int)

    operator fun set(items: List<Item>, preItemCount: Int, adapterNotifier: IAdapterNotifier?)

    fun setNewList(items: List<Item>, notify: Boolean)

    fun addAll(position: Int, items: List<Item>, preItemCount: Int)

    operator fun get(position: Int): Item?

    fun peek(position: Int): Item? {
        return get(position)
    }
}
